### PR TITLE
[GH-1824] Add zone toggle button

### LINKED ISSUE #1824: [FEATURE] Add button to deactivate the zone
## 🚀 Feature

Add a button to deactivate the zone. It will be under "dev" tab.

### PR DESCRIPTION
**Pull request by Nico-Sanchez**
> [!IMPORTANT]  
> This PR must be merged along with https://github.com/lambdaclass/mirra_backend/pull/688

Closes #1824 

## Motivation

We need several tools for testing purposes. This is one of them.
Deactivating the zone, we can last longer.

## Summary of changes

- [Add zone toggle button](https://github.com/lambdaclass/champions_of_mirra/pull/1827/commits/21155d4fc5c7a233c7eff98f801353db535380f3)
- [Add new ToggleZone protobuf message](https://github.com/lambdaclass/champions_of_mirra/pull/1827/commits/83e6534d0baf58939755f8606c0e9075c009f993)
- [Send ToggleZone message when interacting with UI button](https://github.com/lambdaclass/champions_of_mirra/pull/1827/commits/61758ff0c550f8211991cc04e49cecb189068a5a)
- [Add text below zone button to show if its on/off](https://github.com/lambdaclass/champions_of_mirra/pull/1827/commits/917a6a423cb401a43d5cb900966fe1f797dd89f8)

> [!NOTE]  
> There's a known issue where the button starts on but the icon is not highlighted, It should be fixed in a future PR.

## How has this been tested?

Just play a game and toggle zone in settings > dev tab.
Make sure you use the following backend branch -> https://github.com/lambdaclass/mirra_backend/pull/688

## Checklist
- [x] I have tested the changes locally.
- [x] I have tested the whole game after applying the changes, not only the affected areas.
- [x] I self-reviewed the changes on GitHub, line by line.
- [x] I have tested the changes in another devices.
  - [ ] Tested in iOS.
  - [x] Tested in Android.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

File: client/Assets/Scripts/GameServerConnectionManager.cs [hunk 1]
```diff
@@ -256,6 +256,13 @@ public void SendUseItem(long timestamp)
         SendGameAction(gameAction);
     }
 
+    public void SendToggleZone(long timestamp)
+    {
+        ToggleZone toggleZone = new ToggleZone { Zone = "toggle" };
```
Comment (agustinesco) [inline client/Assets/Scripts/GameServerConnectionManager.cs:261]: Can't you send an empty struct?

File: client/Assets/Scripts/UI/Settings/ToggleButton.cs [hunk 2]
```diff
@@ -84,6 +86,26 @@ public void ToggleClientPredictionGhost()
...
+        }
+        else
+        {
+            if (textState != null)
+            {
+                textState.text = "Off";
+            }
+        }
+        ToggleUIState(zoneActive);
+    }
```
Comment (agustinesco) [inline client/Assets/Scripts/UI/Settings/ToggleButton.cs:107]: You can rewrite this if to:
```suggestion
        public void ToggleZone()
    {
        bool zoneActive = battle.zoneActive;
        if (textState != null) {
            if(zoneActive){
                textState.text = "On";
            } else {
                textState.text = "Off";
            }
        }
        ToggleUIState(zoneActive);
    }
```

Comment (Nico-Sanchez) [reply client/Assets/Scripts/UI/Settings/ToggleButton.cs:107]: I could change this but I'll delete it in a future PR.

File: client/Assets/Scripts/GameServerConnectionManager.cs (hunk 1 above)
Comment (Nico-Sanchez) [reply client/Assets/Scripts/GameServerConnectionManager.cs]: Sure! Changed!

**Review by agustinesco — APPROVED** [review]
LGTM

**Pull request merged**